common/usb_cdc.c

```diff
@@ -550,6 +550,8 @@ void AT91F_CDC_Enumerate() {
 			AT91F_USB_SendData(pUdp, devDescriptor, MIN(sizeof(devDescriptor), wLength));
 		else if (wValue == 0x200)  // Return Configuration Descriptor
 			AT91F_USB_SendData(pUdp, cfgDescriptor, MIN(sizeof(cfgDescriptor), wLength));
+		else if ((wValue & 0x300) == 0x300)  // Return Manufacturer Descriptor - this is needed by Android

```

Comment:
This would override all get stringdescriptor calls (next lines)

Reply:
Ok. so this code "as is" is not the answer. I'll do some testing to find a change which works for both cases